kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -283,18 +246,15 @@ private static InvalidClusterException filterResourceNotFound(VirtualKafkaCluste
         return new InvalidClusterException(ClusterCondition.filterNotFound(ResourcesUtil.name(cluster), filterRef.getName()));
     }
 
-    @NonNull
-    private static InvalidClusterException targetClusterResourceNotFound(VirtualKafkaCluster cluster) {
-        return new InvalidClusterException(ClusterCondition.targetClusterRefNotFound(ResourcesUtil.name(cluster), cluster.getSpec().getTargetCluster()));
-    }
-
     /**
      * Look up a Filter CR from the group, kind and name given in the cluster.
      */
     @NonNull
-    private static GenericKubernetesResource filterResourceFromRef(VirtualKafkaCluster cluster, Context<KafkaProxy> context, Filters filterRef)
+    private static GenericKubernetesResource filterResourceFromRef(VirtualKafkaCluster cluster,
```

Comment:
I wonder if these find methods should be pushed to the ResolutionResult.   Presumably InvalidClusterException would now be completely unexpected because we should never be building a config for a cluster that's not resolved.

Reply:
I wonder if building up the resolved graph should be part of the result. Then clients wouldn't need to lookup references.

Would leave it for another PR though.